crates/primitives/starknet/src/sequencer_address/mod.rs

```diff
@@ -0,0 +1,116 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::array::TryFromSliceError;
+
+use scale_codec::{Decode, Encode};
+use sp_inherents::{InherentData, InherentIdentifier, IsFatalError};
+use thiserror_no_std::Error;
+
+/// The identifier for the `sequencer_address` inherent.
+pub const INHERENT_IDENTIFIER: InherentIdentifier = *b"seqaddr0";
+
+/// Default value in case the sequencer address is not set.
+pub const DEFAULT_SEQUENCER_ADDRESS: [u8; 32] =
+    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 222, 173];
+
+/// The storage key for the sequencer address value.
+pub const SEQ_ADDR_STORAGE_KEY: &[u8] = b"starknet::seq_addr";
+
+/// The inherent type for the sequencer address.
+pub type InherentType = [u8; 32];
+
+#[derive(Decode, Encode, Error, sp_runtime::RuntimeDebug)]
+/// Error types when working with the sequencer address.
+pub enum InherentError {
+    /// Submitted address must be `[u8; 32]`.
+    #[error("Inherent decoding error")]
+    WrongAddressFormat,
+}
+
+impl IsFatalError for InherentError {
+    fn is_fatal_error(&self) -> bool {
+        match self {
+            InherentError::WrongAddressFormat => true,
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl InherentError {
+    /// Try to create an instance ouf of the given identifier and data.
+    pub fn try_from(id: &InherentIdentifier, mut data: &[u8]) -> Option<Self> {
+        if id == &INHERENT_IDENTIFIER { <InherentError as Decode>::decode(&mut data).ok() } else { None }
+    }
+}
+
+/// Auxiliary trait to extract sequencer address inherent data.
+pub trait SequencerAddressInherentData {
+    /// Get sequencer address inherent data.
+    fn sequencer_address_inherent_data(&self) -> Result<Option<InherentType>, sp_inherents::Error>;
+}
+
+impl SequencerAddressInherentData for InherentData {
+    fn sequencer_address_inherent_data(&self) -> Result<Option<InherentType>, sp_inherents::Error> {
+        self.get_data(&INHERENT_IDENTIFIER)
+    }
+}
+
+#[derive(Copy, Clone, Decode, Encode, sp_runtime::RuntimeDebug)]
+#[cfg(feature = "std")]
```

Comment:
Rather than putting every symbol down this line behind `#[cfg(feature = "std")]` individually, you should create a module, put it behind `#[cfg(feature = "std")]`, and pub use everything in this module if `#[cfg(feature = "std")]`